crypto/x509/v3_utl.c

```diff
@@ -686,10 +686,40 @@ typedef int (*equal_fn)(const unsigned char *pattern, size_t pattern_len,
                         const unsigned char *subject, size_t subject_len,
                         unsigned int flags);
 
+// Skip pattern prefix to match "wildcard" subject
+static void skip_prefix(const unsigned char **p, size_t *plen,
+                        size_t subject_len, unsigned int flags) {
+  const unsigned char *pattern = *p;
+  size_t pattern_len = *plen;
+
+  // If subject starts with a leading '.' followed by more octets, and
+  // pattern is longer, compare just an equal-length suffix with the
+  // full subject (starting at the '.'), provided the prefix contains
+  // no NULs.
+  if ((flags & _X509_CHECK_FLAG_DOT_SUBDOMAINS) == 0) {
+    return;
+  }
+
+  while (pattern_len > subject_len && *pattern) {
+    if ((flags & X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS) && *pattern == '.') {
```

Comment:
This condition can [never be true](https://github.com/aws/aws-lc/blob/main/include/openssl/x509.h#L5029):
```
#define X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS 0
```

Are you missing an update to the header file?